_posts/pl/2024-06-28-czy-wiesz-jak-uzywac-splice-w-javascript.md

```diff
@@ -0,0 +1,53 @@
+---
+layout:    post
+title:     Czy wiesz, jak używać splice w JavaScript?
+description: Splice to metoda dostępna dla tablicy. Pozwala usuwać, zmienić istniejący element lub dodać nowy element do tablicy.
+date:      2024-06-28T08:00:00+01:00
+published: true
+didyouknow: true
+lang: pl
+author: dmejer
+image: /assets/img/posts/2024-06-28-czy-wiesz-jak-uzywac-splice-w-javascript/code.webp
+tags:
+- javascript
+---
+`Splice` to metoda dostępna dla tablicy. Pozwala usuwać, zmienić istniejący element lub dodać nowy element do tablicy.
+
+Przykład usunięcia elementów:
+```javascript
+const a = [1, 2, 3, 4];
+const b = a.splice(1, 1); // usunięcie 1 elementu od 1 indeksu
```

Comment:
to "usunięcie OD indeksu" normalnie bym zmieniła na "usunięcie 1 elementu Z indeksu"
ale może jakiś związek mi tu umyka :)

Reply:
Indeks to taki numer porządkowy w tablicy i "z indeksu" w takim kontekście nie pasuje, ale faktycznie szyk zdania jest zły powinno być raczej "indeksu 1" a nie "1 indeksu" - w domyśle to zdanie powinno brzmieć mniej więcej: "usunięcie jednego elementu tablicy zaczynając od indeksu jeden"